core/src/main/java/org/infinispan/stats/impl/CacheContainerStatsImpl.java

```diff
@@ -64,7 +64,11 @@ public void resetStatistics() {
       if (getStatisticsEnabled()) {
          for (String cn : cm.getCacheNames()) {
             if (cm.cacheExists(cn)) {
-               cm.getCache(cn).getAdvancedCache().getStats().reset();
+               try {
+                  cm.getCache(cn).getAdvancedCache().getStats().reset();
+               } catch (SecurityException e) {
+                  // no-op, skipping this cache
```

Comment:
> I was thinking that we could use a proper user
I'm not sure what actually you mean by proper user. If one cache has user A as an admin and another cache user B, who is the proper user in this case? As mentioned above, this can be solved by {{SecurityActions}} (i.e. skipping security checks), but I'm really not sure we want this, IMHO user which hasn't access to the cache should see its statistics at all and shouldn't be even aware such cache exists. Anyway, lets wait what @tristantarrant thinks about it.